modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/editor/Editor.js

```diff
@@ -13,15 +13,54 @@
  */
 
 import CKEditor from 'ckeditor4-react';
-import React from 'react';
+import React, {useEffect, useState} from 'react';
 
-const BASEPATH = '/o/frontend-editor-ckeditor-web/ckeditor/';
+const CKEDITOR_BASEPATH = '/o/frontend-editor-ckeditor-web/ckeditor/';
+const CKEDITOR_PATH = `${CKEDITOR_BASEPATH}ckeditor.js`;
 
 const Editor = React.forwardRef((props, ref) => {
-	return <CKEditor ref={ref} {...props} />;
+	const [initialized, setInitialized] = useState(!!window.CKEDITOR);
+
+	useEffect(() => {
+		let script;
+
+		const destroyGlobalCkEditor = () => {
+			if (
+				window.CKEDITOR &&
+				Object.keys(window.CKEDITOR.instances).length === 0
+			) {
+				window.CKEDITOR = undefined;
+
+				Liferay.detach('beforeScreenFlip', destroyGlobalCkEditor);
+			}
+		};
+
+		if (!initialized) {
+			script = document.createElement('script');
+
+			script.setAttribute('data-senna-track', 'temporary');
+			script.src = CKEDITOR_PATH;
+
+			script.onload = () => {
+				setInitialized(true);
+			};
```

Comment:
As this is async, probably want to guard the `setInitialized()` call in an `isMounted()` check from the `useIsMounted` hook.